geode-assembly/src/acceptanceTest/java/org/apache/geode/cache/wan/SeveralGatewayReceiversWithSamePortAndHostnameForSendersTest.java

```diff
@@ -214,6 +215,56 @@ public void testTwoSendersWithSameIdShouldUseSameValueForEnforceThreadsConnectTo
 
   }
 
+
+  /**
+   * The aim of this test is verify that when several gateway receivers in a remote site share the
+   * same port and hostname-for-senders, the pings sent from the gateway senders reach the right
+   * gateway receiver and not just any of the receivers. Check that only one destination will be
+   * pinged.
+   */
+  @Test
+  public void testPingsToReceiversWithSamePortAndHostnameForSendersReachTheRightReceiver()
+      throws InterruptedException {
+    String senderId = "ln";
+    String regionName = "region-wan";
+    final int remoteLocPort = docker.getExternalPortForService("haproxy", 20334);
+
+    int locPort = createLocator(VM.getVM(0), 1, remoteLocPort);
+
+    VM vm1 = VM.getVM(1);
```

Comment:
I notice that this is a dunit test in the `acceptanceTest` src set. It should probably be in `distributedTest` instead. The acceptance tests are all written using GfshRule without dunit.

Test names should reflect the test types:

Unit test: MyTest

- Integration test: `MyIntegrationTest`
- Distributed test: `MyDistributedTest`
- Acceptance test: `MyAcceptanceTest`

Reply:
Thanks for comments. Since this problem can be reproduced only with docker, this was only test to prove improvements.
I will add UT for added functions, classes.